hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/impl/ITestS3APutIfMatch.java

```diff
@@ -85,7 +115,7 @@ public void testPutIfAbsentConflict() throws IOException {
           Assert.assertEquals(RemoteFileChangedException.class, e.getClass());
 
           S3Exception s3Exception = (S3Exception) e.getCause();
-          Assert.assertEquals(s3Exception.statusCode(), 412);
+          Assertions.assertThat(s3Exception.statusCode()).isEqualTo(412);
```

Comment:
There are enough checks of status codes in this test suite it can be pulled out. and ideally, the exception with its stack trace not lost. "every stack trace is sacred".

```
assertS3ExceptionStatusCode(int code, S3Exception ex) {
 if (ex.statusCode() != code) {
   throw new AssertionError("Expected status code " + code + " from " + ex,
     ex)
  }
}
```
This would raise an assertion error showing what was expected, but retain that original stack. And it could be used throughout the suite.

Reply:
done.